### PR TITLE
Get fields

### PR DESCRIPTION
**Pull request by domtra**

### CODE REVIEW

**Comment by emcarru** [comment]
There's quite some references to using `registerFields` in the docs, but I can strip this all out if we merge this.

File: Components/BlockCollapse/functions.php [hunk 1]
```diff
@@ -2,10 +2,9 @@
 
 namespace Flynt\Components\BlockCollapse;
 
-use Flynt\Api;
-
-Api::registerFields('BlockCollapse', [
-    'layout' => [
+function getLayout()
```
Comment (steffenbew) [inline Components/BlockCollapse/functions.php]: 'layout' as a name makes sense to me when used in a pure ACF context.
As a function name on a component I would rather expect something like `getFields`.

**Comment by domtra** [comment]
@emcarru yes, there is. i intentionally left this open for you to take care of, if we merge this.

File: Components/BlockCollapse/functions.php (hunk 1 above)
Comment (domtra) [reply Components/BlockCollapse/functions.php]: This is strictly for getting the layout. you might want to create other functions to get other fields. that is why i gave it a specific name, not a general one. it does make more sense to me like this. `getFields` could be any kind of field, or an array of all fields where is not really clear which ones.

Comment (steffenbew) [reply Components/BlockCollapse/functions.php]: I understand that there might be different field groups. But I don't understand why this array of fields would be the layout of the component. Reading the function call `Components\BlockTextImageCrop\getLayout()`, I expect to get "a layout" returned. And then my head goes: "What is the layout of a component in PHP?" and can't make any sense of it. So to me, it's really just about the lacking context that I believe is currently more implicit.

Comparing to the current call `Api::loadFields('BlockTextImageCrop', 'layout')`, I could also imagine something like `Components\BlockTextImageCrop\getFields('layout')` or `Components\BlockTextImageCrop\getACFLayout()` or `Components\BlockTextImageCrop\getLayoutFields()`.

Comment (domtra) [reply Components/BlockCollapse/functions.php]: i would prefer the `getACFLayout` then. i can see what you mean. the thing is, `getLayout` does not return an array of fields, but one layout field where the sub_fields are what we see as fields of the component.

Comment (steffenbew) [reply Components/BlockCollapse/functions.php]: As long as it's specific and a recurring concept (like naming all ACF related functions "getACFSomething"), it's fine with me.

Comment (domtra) [reply Components/BlockCollapse/functions.php]: @steffenbew updated

**Pull request merged**